cms/djangoapps/contentstore/tests/test_course_listing.py

```diff
@@ -107,6 +109,88 @@ def test_errored_course_global_staff(self):
+        """
+        Create courses and compare the time to fetch accessible courses for the staff user and
+        normal user. When user is staff, fetched courses contains light weight course summery objects and
+        contains CourseDescriptor objects otherwise. Fetching courses as a staff should take lesser time than
+        non-staff.
+        """
+
+        # Add user to the staff & create courses
+        GlobalStaff().add_users(self.user)
+        for num in range(USER_COURSES_COUNT):
+            self._create_course_with_access_groups(
+                course_location=CourseLocator('Org', 'CreatedCourse' + str(num), 'Run'),
+                user=self.user,
+                store=default_store
+            )
+
+        # Measure time to fetch courses for staff user
+        with Timer() as staff_course_fetch_time:
+            courses_list_for_staff, __ = get_courses_accessible_to_user(self.request)
+
+        # Remove staff role of the user & measure courses access time.
+        GlobalStaff().remove_users(self.user)
+        with Timer() as user_course_fetch_time:
+            courses_list_by_user, __ = get_courses_accessible_to_user(self.request)
+
+        # Verify returned courses as a staff & as non-staff are equal
+        self.assertEqual(len(courses_list_for_staff), len(courses_list_by_user), 'Returned Courses should be equal')
+
+        # Verify that course access is faster for the staff user
+        self.assertGreater(
+            user_course_fetch_time.elapsed,
+            staff_course_fetch_time.elapsed,
+            'Course iteration time for staff should be less'
+        )
+
+    @ddt.data(ModuleStoreEnum.Type.split, ModuleStoreEnum.Type.mongo)
+    def test_staff_course_listing(self, default_store):
+        """
+        Test that the fetch accessible courses list for staff and non-staff when get_course returns
+        CourseSummary instances for staff and CourseDescriptor for non-staff users.
+        """
+
+        # Assign & verify staff role to the user
+        GlobalStaff().add_users(self.user)
+        self.assertTrue(GlobalStaff().has_user(self.user))
+
+        # Create few courses
+        for num in range(USER_COURSES_COUNT):
+            self._create_course_with_access_groups(
+                course_location=CourseLocator('Org', 'CreatedCourse' + str(num), 'Run'),
+                user=self.user,
+                store=default_store
+            )
+
+        # Fetch accessible courses list & verify their count
+        courses_list_by_staff, __ = get_courses_accessible_to_user(self.request)
+        self.assertEqual(len(courses_list_by_staff), USER_COURSES_COUNT)
+
+        # Verify fetched accessible courses list is a list of CourseSummery instances
+        is_course_summary_descriptors = all(
+            [isinstance(course, CourseSummary)for course in courses_list_by_staff]
+        )
+        self.assertTrue(
+            is_course_summary_descriptors,
```

Comment:
It might be clearer to just put the `all` statement here:

```python
self.assertTrue(
    all(isinstance(course, CourseSummary) for course in courses_list_by_staff)
)